app/src/main/java/com/lla/panicbutton/ui/fragments/onboardingfragments/ViewPagerFragment.kt

```diff
@@ -0,0 +1,79 @@
+package com.lla.panicbutton.ui.fragments.onboardingfragments
+
+import android.content.Context
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager2.widget.ViewPager2
+import com.lla.panicbutton.R
+import com.lla.panicbutton.adapters.ViewPagerAdapter
+import com.tbuonomo.viewpagerdotsindicator.SpringDotsIndicator
+import kotlinx.android.synthetic.main.fragment_view_pager.view.*
+
+class ViewPagerFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_view_pager, container, false)
+
+        val fragmentList = arrayListOf<Fragment>(
+            Welcome2Fragment(),
+            Welcome3Fragment(),
+            Welcome4Fragment()
+        )
+
+        val adapter = ViewPagerAdapter(
+            fragmentList,
+            requireActivity().supportFragmentManager,
+            lifecycle
+        )
+
+        view.viewPager2.adapter = adapter
+
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val fragmentList = arrayListOf<Fragment>(
+            Welcome2Fragment(),
+            Welcome3Fragment(),
+            Welcome4Fragment()
+        )
+
+        val adapter = ViewPagerAdapter(
+            fragmentList,
+            requireActivity().supportFragmentManager,
+            lifecycle
+        )
+
+        view.viewPager2.adapter = adapter
+
+        val springDotsIndicator = requireView().findViewById<SpringDotsIndicator>(R.id.spring_dots_indicator)
+        val viewPager = requireView().findViewById<ViewPager2>(R.id.viewPager2)
+        springDotsIndicator?.setViewPager2(viewPager)
+
+        view.viewPagerNextButton.setOnClickListener{
+            if (viewPager.currentItem+1 < adapter.itemCount){
+                viewPager.currentItem+=1
+            }else{
+                findNavController().navigate(R.id.action_viewPagerFragment_to_setupFragment)
+                onBoardingFinished()
+            }
+        }
+    }
+
+    private fun onBoardingFinished(){
```

Comment:
maybe endOnboarding?

Reply:
okay, i'll change this too

Reply:
however, it's more of an indicator than an actual action, no?

Reply:
I feel like all functions that return void are actions, so here you're calling it when you want to set nextButton4 to false and by doing that, you're ending onboarding. So you use this to see if onboarding is finished, right? and you also use "setUpNext" to also see if onboarding is finished, can't you have just one setting that checks that? Maybe I'm misunderstanding them. but when the user presses the arrow button on the upload function (which is the last step in onboarding) we could set "isFirstTime" to false and use that everywhere to see.

Reply:
Ohh I see what you mean. Putting it into one shared preference for everything that is included in onboarding. I think I had them separately because of how the viewpager was. Let me try with only a single setting.